stub/src/main/java/io/grpc/stub/StreamObserver.java

```diff
@@ -68,6 +68,12 @@
    * exception is thrown by an implementation of {@code onError} no further calls to any method are
    * allowed.
    *
+   * <p>{@code t} should be a {@link io.grpc.StatusException} or {@link
+   * io.grpc.StatusRuntimeException}, but other {@code Throwable} types are possible. Callers should
```

Comment:
We're making the exception because this is a clone of Rx's Observer and the application may be mixing them. The library is always emitting `StatusRuntimeException`. If we were going to remove the exception, I think the method signature itself would change.